internal/util/projutil/project_util.go

```diff
@@ -140,7 +152,52 @@ func getHomeDir() (string, error) {
 // under $GOPATH and returns the current directory's import path,
 // e.g: "github.com/example-inc/app-operator"
 func CheckAndGetProjectGoPkg() string {
-	gopath := MustSetGopath(MustGetGopath())
+	gopath := MustSetFirstGopath(MustGetGopath())
+	return parseGoPkg(gopath)
+}
+
+// GetGoPkg returns the current directory's import path by parsing it from
+// wd if this project's repository path is rooted under $GOPATH/src, or
+// from go.mod the project uses go modules to manage dependencies.
+//
+// Example: "github.com/example-inc/app-operator"
+func GetGoPkg() string {
+	// Default to reading from go.mod, as it should usually have the (correct)
```

Comment:
More of a general question. Do we currently use `CheckAndGetProjectGoPkg` and will we use `GetGoPkg` to get arbitrary package paths within a project, or is it only ever intended to get the root project path?

It looks like both functions will return arbitrary paths if the project is in $GOPATH. And since projects can have `go.mod` files in various subtrees, `GetGoPkg` can return paths outside of the project root based on `go.mod` files as well.

Just asking because we should either explicitly run `CheckProjectRoot` or make sure that neither the function name (e.g. `CheckAndGetProjectGoPkg`) nor the GoDoc comments imply that these functions return the root project directory.

Reply:
We do not have an explicit check that `CheckAndGetProjectGoPkg()` is used in the project root, but I think we expect it to be based on how the SDK uses that function. Problems come up if we want to allow specific subcommands of `operator-sdk` to be run outside of the project root, ex. a user with a non-standard project structure wants to build a CSV. IMO it should be usable globally; if we want the root package we should call `MustInProjectRoot()`.

We can use the root `go.mod` only to solve the multiple modfile issue, and defer to the `GOPATH` parsing code if the cwd's `go.mod` is not root. There should always be a root modfile. We can also append the cwd to the go package returned if a modfile is used to be consistent with `CheckAndGetProjectGoPkg()` behaviour.

Reply:
> We can use the root go.mod only to solve the multiple modfile issue, and defer to the GOPATH parsing code if the cwd's go.mod is not root. There should always be a root modfile. We can also append the cwd to the go package returned if a modfile is used to be consistent with CheckAndGetProjectGoPkg() behaviour.

We may not have to do this. In the examples I've seen, if a single repo has multiple `go.mod` files, all of the `go.mod` files still contain their full module path, so it would be fine to just use whatever is in the CWD, I think.

There may be some extremely rare cases where this breaks for users that have a monorepo with multiple projects, each with multiple `go.mod` files, where the filesystem path doesn't line up with the module path. But let's cross that bridge if/when we get there.